infra/security.tf

```diff
@@ -52,7 +51,35 @@ module "iam_assumable_role_with_oidc" {
 
   number_of_role_policy_arns = 1
 
-  role_policy_arns = [module.iam_policy.arn]
+  role_policy_arns = [module.iam_policy.arn, "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy"]
 
   oidc_fully_qualified_audiences = ["sts.amazonaws.com"]
 }
+
+module "iam_eks_role" {
+  source    = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  role_name = "jovand-irsa-lb"
+
+  attach_load_balancer_controller_policy = true
+
+  oidc_providers = {
+    main = {
+      provider_arn               = module.eks.oidc_provider_arn
+      namespace_service_accounts = ["default"]
```

Comment:
You are missing the name of the svc account that pod will use, this is only namespace which is also required.